addons/xeh_a3/CfgEventHandlers.hpp

```diff
@@ -159,20 +159,26 @@ class Extended_init_Eventhandlers {
         class C_Kart_01_F /* : C_Kart_01_F_Base  */{
                 SLX_BIS = "if (local (_this select 0)) then {[(_this select 0), """", nil, nil] call bis_fnc_initVehicle; [(_this select 0)] call bis_fnc_initVehicleKart;};";
         };
+        class Snake_random_F /* : Animal_Base_F */ {
+                SLX_BIS = "_FRnd = compile preprocessFile ""\A3\animals_f\Snakes\scripts\randomize.sqf"";(_this select 0) call _FRnd;";
+        };
         class Fin_random_F /* : Fin_Base_F */ {
-                SLX_BIS = "_FRnd = compile loadFile ""\A3\animals_f_beta\dog\scripts\randomize.sqf"";(_this select 0) call _FRnd;";
+                SLX_BIS = "_FRnd = compile preprocessFile ""\A3\animals_f_beta\dog\scripts\randomize.sqf"";(_this select 0) call _FRnd;";
         };
         class Alsatian_Random_F /* : Alsatian_Base_F */ {
-                SLX_BIS = "_FRnd = compile loadFile ""\A3\animals_f_beta\dog\scripts\randomize_als.sqf"";(_this select 0) call _FRnd;";
+                SLX_BIS = "_FRnd = compile preprocessFile ""\A3\animals_f_beta\dog\scripts\randomize_als.sqf"";(_this select 0) call _FRnd;";
         };
         class Goat_random_F /* : Goat_Base_F */ {
-                SLX_BIS = "_FRnd = compile loadFile ""\A3\animals_f_beta\goat\scripts\randomize.sqf"";(_this select 0) call _FRnd;";
+                SLX_BIS = "_FRnd = compile preprocessFile ""\A3\animals_f_beta\goat\scripts\randomize.sqf"";(_this select 0) call _FRnd;";
         };
         class Sheep_random_F /* : Animal_Base_F */ {
-                SLX_BIS = "_FRnd = compile loadFile ""\A3\animals_f_beta\Sheep\scripts\randomize.sqf"";(_this select 0) call _FRnd;";
+                SLX_BIS = "_FRnd = compile preprocessFile ""\A3\animals_f_beta\Sheep\scripts\randomize.sqf"";(_this select 0) call _FRnd;";
         };
-        class Snake_random_F /* : Animal_Base_F */ {
-                SLX_BIS = "_FRnd = compile loadFile ""\A3\animals_f\Snakes\scripts\randomize.sqf"";(_this select 0) call _FRnd;";
+        class Cock_random_F /* : Fowl_Base_F */ {
+                SLX_BIS = "_FRnd = compile preprocessFile ""\A3\animals_f_beta\Chicken\scripts\randomize.sqf"";(_this select 0) call _FRnd;";

```

Comment:
Cock_random_F is not randomized in 1.48 (it does not have an init event handler). We can still add it into a pre-A3 1.50 XEH release, but with a forward-looking XEH init event handler - the one that the animal will have in 1.50: ```(_this select 0) call bis_fnc_animalRandomization;```

That way it will do nothing in 1.48 and work as intended in 1.50.